### PR TITLE
Adding parsing for collection memberships for the boxItem model

### PR DESCRIPTION
**Pull request by jdizoglio**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jdizoglio has signed the CLA. Thanks for the pull request!

**Comment by jdizoglio** [comment]
Example response:
{
   “type” : “file”,
   “id” : “123,
   “collections” : [
       {
           “type”: “collection”,
           “id”: “555”,
           “name”: “Favorites”,
           “collection_type”: “favorites”,
       },
       {
           “type”: “collection”,
           “id”: “666”,
           “name”: “Another collection with ranking”,
           “collection_type”: “other”
       }
   ]
   “collection_memberships” : [
       {
           “type” : “collection_membership”
           “collection” : {
               “type”: “collection”,
               “id”: “555”,
               “name”: “Favorites”,
               “collection_type”: “favorites”,
           },
           “rank” : 1111111111
       },
       {
           “type” : “collection_membership”
           “collection” : {
               “type”: “collection”,
               “id”: “666”,
               “name”: “Another collection with ranking”,
               “collection_type”: “other”
               },
               “rank” : 22222222222
           }
       }
   ]
}

File: BoxContentSDK/BoxContentSDK/Models/BOXCollection.m [hunk 1]
```diff
@@ -29,7 +29,7 @@ - (instancetype)initWithJSON:(NSDictionary *)JSONResponse
         self.collectionRank = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyCollectionRank
                                                              inDictionary:JSONResponse
                                                           hasExpectedType:[NSNumber class]
-                                                              nullAllowed:NO];
+                                                              nullAllowed:YES];
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Models/BOXCollection.m]: nil instead of NSNull

Comment (jdizoglio) [reply BoxContentSDK/BoxContentSDK/Models/BOXCollection.m]: you are correct, done.

File: BoxContentSDK/BoxContentSDK/BOXContentSDKConstants.h [hunk 2]
```diff
@@ -329,6 +329,7 @@ extern NSString *const BOXAPIObjectKeyExtension;
 extern NSString *const BOXAPIObjectKeyIsPackage;
 extern NSString *const BOXAPIObjectKeyAllowedSharedLinkAccessLevels;
 extern NSString *const BOXAPIObjectKeyCollections;
+extern NSString *const BOXAPIObjectKeyCollectionMemberships;
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/BOXContentSDKConstants.h:332]: need to add this to the get all fields parameter as well

Comment (crousselle) [inline BoxContentSDK/BoxContentSDK/BOXContentSDKConstants.h:332]: isn't that supposed to be a private field ? if it does not appear on our dev page, probably shouldn't be available to devs

Comment (boxtcarpel) [reply BoxContentSDK/BoxContentSDK/BOXContentSDKConstants.h:332]: It is public but undocumented, it's ok for all consumers of the Box iOS SDK

Comment (jdizoglio) [reply BoxContentSDK/BoxContentSDK/BOXContentSDKConstants.h:332]: done

Comment (jdizoglio) [reply BoxContentSDK/BoxContentSDK/BOXContentSDKConstants.h:332]: We need access to the property. I don't see that we have precedence for  a private constant.

File: BoxContentSDK/BoxContentSDK/Models/BOXItem.m [hunk 3]
```diff
@@ -239,6 +240,34 @@ - (instancetype)initWithJSON:(NSDictionary *)JSONResponse
...
+                collection.collectionRank = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyCollectionRank
+                                                                           inDictionary:membershipDictionary
+                                                                        hasExpectedType:[NSNumber class]
+                                                                            nullAllowed:YES];
+                
+                [collections addObject:collection];
+            }
+        }
+        
+        // Set all collections and collection memberships found in json response.
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Models/BOXItem.m:270]: Do we want collections and collection memberships in a single array? any potential issues that may cause?

**Review by boxtcarpel — APPROVED** [review]

File: BoxContentSDK/BoxContentSDK/Models/BOXItem.m (hunk 3 above)
Comment (jdizoglio) [reply BoxContentSDK/BoxContentSDK/Models/BOXItem.m:270]: Talked offline. The client will handle this case.

**Pull request merged**